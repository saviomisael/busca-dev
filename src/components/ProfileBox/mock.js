import Image from '../../assets/images/person_white.svg';

export const profileBoxPropsMock = {
  avatar: Image,
  username: 'johndoe',
  fullname: 'John Doe',
  bio: 'bio Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
  blog: 'blog.com',
};
