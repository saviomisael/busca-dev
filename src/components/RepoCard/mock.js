export const repoCardPropsMock = {
  repoName: 'repo name',
  description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  starsCount: 0,
  repoUrl: 'github.com/repo',
};
