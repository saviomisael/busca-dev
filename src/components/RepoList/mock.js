export const repoListPropsMock = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
  ],
};

export const manyRepositories = {
  repositories: [
    {
      repoId: 1,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 2,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 3,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 3,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 0,
      repoUrl: 'github.com/repo',
    },
    {
      repoId: 4,
      repoName: 'repo name',
      description: 'Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.',
      starsCount: 5,
      repoUrl: 'github.com/repo',
    },
  ],
};
