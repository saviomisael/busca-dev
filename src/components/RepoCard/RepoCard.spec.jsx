import { screen } from '@testing-library/react';
import { RepoCard } from '.';
import { repoCardPropsMock } from './mock';

describe('<RepoCard />', () => {
  it('should renders component with default props', () => {
    const props = { ...repoCardPropsMock };

    delete props.description;

    global.renderTheme(<RepoCard {...props} />);

    const descriptionElement = screen.queryByText(/Repository Description lorem/i);

    expect(descriptionElement).not.toBeInTheDocument();
  });

  it('should renders description when description prop is truthy', () => {
    global.renderTheme(<RepoCard {...repoCardPropsMock} />);

    const descriptionElement = screen.getByText(/Repository Description lorem/i);

    expect(descriptionElement).toBeInTheDocument();
  });

  it('should matches with snapshot', () => {
    const { container } = global.renderTheme(<RepoCard {...repoCardPropsMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
