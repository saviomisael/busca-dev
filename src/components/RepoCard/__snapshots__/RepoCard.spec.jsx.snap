// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> should matches with snapshot 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 0.7rem;
  column-gap: 0.7rem;
}

.c6 {
  text-transform: uppercase;
  font-size: 1.6rem;
  color: #284B63;
}

.c5 {
  -webkit-filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
  filter: invert(23%) sepia(6%) saturate(4839%) hue-rotate(162deg) brightness(93%) contrast(82%);
}

.c0 {
  padding: 1.5rem;
  border-radius: 0.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  row-gap: 1rem;
  background-color: #D9D9D9;
}

.c0 > * {
  font-size: 1.8rem;
  color: #353535;
}

.c0 > .c3 {
  margin-top: 1rem;
}

.c1 {
  font-weight: 600;
}

.c2 {
  font-weight: 500;
}

<article
  class="c0"
>
  <a
    class="c1"
    href="github.com/repo"
    target="_blank"
  >
    repo name
  </a>
  <p
    class="c2"
  >
    Repository Description lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <div
    class="c3 c4"
  >
    <img
      alt="star icon"
      class="c5"
      src="star_empty.svg"
    />
    <p
      class="c6"
    >
      0
       Estrelas
    </p>
  </div>
</article>
`;
